Makefile

```diff
@@ -37,28 +49,60 @@ cluster-destroy:
 
 .PHONY: spark-build
 spark-build:
-	if [[ -z  "$(shell docker images -q $(SPARK_IMAGE_NAME_WITH_TAG))" ]]; then
-		docker build -t ${SPARK_IMAGE_NAME_WITH_TAG} ${SPARK_IMAGE_DOCKERFILE_PATH}
+	if [[ -z  "$(shell docker images -q $(SPARK_IMAGE_FULL_NAME))" ]]; then
+		docker build \
+			-t ${SPARK_IMAGE_FULL_NAME} \
+			-f ${SPARK_IMAGE_DOCKERFILE_PATH}/Dockerfile \
+			${SPARK_IMAGE_DOCKERFILE_PATH}
 	fi
-	echo "${SPARK_IMAGE_NAME_WITH_TAG}" > $@
+	echo "${SPARK_IMAGE_FULL_NAME}" > $@
 
 .PHONY: docker-build
 docker-build: spark-build
 docker-build:
-	docker build \
-		--build-arg SPARK_IMAGE=$(shell cat spark-build) \
-		-t ${SPARK_ON_K8S_OPERATOR_IMAGE_NAME} \
-		-f ${SPARK_ON_K8S_OPERATOR_DOCKERFILE_PATH} ${SPARK_OPERATOR_DIR}
+	if [[ -z "$(shell docker images -q $(SPARK_ON_K8S_OPERATOR_IMAGE_FULL_NAME))" ]]; then
```

Comment:
it's a neat move but don't we want to rely on `make` here and just `echo` the image name to the target file? Also, it doesn't look like Docker-related targets should be `PHONY`, we can introduce `clean-docker` target for the rebuild purposes.